src/CodeGen_C.cpp

```diff
@@ -1653,22 +1657,35 @@ void CodeGen_C::compile(const Buffer<> &buffer) {
     // used to store stateful module information in offloading runtimes.
     bool is_constant = buffer.dimensions() != 0;
 
-    // Emit the data
-    stream << "static " << (is_constant ? "const" : "") << " uint8_t " << name << "_data[] HALIDE_ATTRIBUTE_ALIGN(32) = {\n";
-    stream << get_indent();
-    for (size_t i = 0; i < num_elems * b.type.bytes(); i++) {
-        if (i > 0) {
-            stream << ",";
-            if (i % 16 == 0) {
-                stream << "\n";
-                stream << get_indent();
-            } else {
-                stream << " ";
+    // If it is an GPU source kernel, we would like to see the actual output, not the
+    // uint8 representation. We use a string literal for this.
```

Comment:
Do all of the APIs emit text? Maybe some of them emit binary data, where this might do problematic things? I'm not sure about Metal for example.

Reply:
Yes, I tested this for OpenCL, Cuda, Metal, D3D12Compute and OpenGLCompute. All print (pretty) GPU kernels.